cpp/src/parquet/encryption_internal.h

```diff
@@ -45,6 +45,9 @@ constexpr int8_t kOffsetIndex = 7;
 /// Performs AES encryption operations with GCM or CTR ciphers.
 class AesEncryptor {
  public:
+  /// Can serve one key length only. Possible values: 16, 24, 32 bytes.
+  explicit AesEncryptor(ParquetCipher::type alg_id, int key_len, bool metadata);
```

Comment:
why isn't static factory sufficient?

Reply:
@emkornfield I want to create a local variable of AesEncryptor, without using smart pointers, so it will store in stack rather than in heap.